File: src/types/receipt.ts
export interface ReceiptItem {
    name: string
    price: number
    quantity: number
    category?: string
}

export interface StoreInfo {
    name: string
    address?: string
}

export interface TotalInfo {
    subtotal: number
    tax: number
    discount: number
    change: number
    amount: number
}

export interface ExtractedData {
    // STORE section
    store: StoreInfo
    // ITEMS section
    items: ReceiptItem[]
    // TOTAL section
    total: TotalInfo
    // OTHERS section
    date: string
    time?: string
    // Additional metadata
    category?: string
    confidence?: number
    rawText: string
    originalResponse?: any // Original API response for feedback
    imageUrl?: string // URL for the receipt image (client-side only)
}

export interface ReceiptData extends ExtractedData {
    id: string
    imageUrls?: string[]
    userId?: string
    createdAt?: string
}

// Legacy flat structure for backward compatibility
export interface LegacyReceiptData {
    id: string
    vendor: string
    address?: string
    date: string
    time?: string
    total: number
    subtotal?: number
    tax?: number
    discount?: number
    change?: number
    items: ReceiptItem[]
    category?: string
    imageUrls?: string[]
    userId?: string
    createdAt?: string
    confidence?: number
    rawText?: string
}

File: backend/app/models/receipt.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, JSON
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from ..db import BaseModel
from datetime import datetime

class Receipt(BaseModel):
    __tablename__ = "receipts"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(String, ForeignKey("users.id"))
    vendor = Column(String)
    address = Column(String, nullable=True)  # Store address
    date = Column(DateTime)
    time = Column(String)
    total = Column(Float)
    subtotal = Column(Float, nullable=True)  # Subtotal amount
    tax = Column(Float)
    discount = Column(Float, nullable=True)  # Discount amount
    change = Column(Float)
    category = Column(String)
    items = Column(JSON)  # Stores array of items
    image_urls = Column(JSON)  # Stores array of image URLs
    confidence = Column(Float)
    created_at = Column(DateTime, default=func.now())
    updated_at = Column(DateTime, default=func.now(), onupdate=func.now())

    user = relationship("User", back_populates="receipts")
